pkg/tfbridge/provider.go

```diff
@@ -661,6 +661,15 @@ func (p *Provider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*pulum
 	doIgnoreChanges(ctx, res.TF.Schema(), res.Schema.Fields, olds, news, req.GetIgnoreChanges(), diff)
 	detailedDiff, changes := makeDetailedDiff(ctx, res.TF.Schema(), res.Schema.Fields, olds, news, diff)
 
+	// There are some providers/situations which `makeDetailedDiff` distorts the expected changes, leading
+	// to changes being dropped by Pulumi.
+	// Until we address https://github.com/pulumi/pulumi-terraform-bridge/issues/1501, it is safer to refer
```

Comment:
This change literally addresses 1501

Reply:
It does; but I'm under the impression we'll want to keep it open so we can fix the bugs in `makeDetailedDiff` "for real".

I'll amend the working to reflect more nuance.

Reply:
I don't think so, I'm happy to close. We can open new issues for makeDetailedDiff referencing the closed issue.